engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -486,6 +486,81 @@ describe('System test: QuerySparql', () => {
           });
         });
       });
+
+      describe('functionArgumentsCache', () => {
+        let query: string;
+        let stringType: RDF.NamedNode;
+        let quads: RDF.Quad[];
+        let store: Store;
+        beforeAll(() => {
+          // The query provided is completely arbitrary and should not change the results of this test
+          // besides the value in the cache.
+          query = `SELECT (strlen(?x) AS ?len) WHERE {
+            ?s ?p ?x
+          }
+          `;
+          stringType = DF.namedNode('http://www.w3.org/2001/XMLSchema#string');
+          store = new Store();
+          quads = [
+            DF.quad(DF.namedNode(':a'), DF.namedNode(':p'), DF.literal('apple', stringType)) ];
+          store.addQuads(quads);
+        });
+
+        it('is used when provided', async() => {
+          const context = <any> { sources: [ store ]};
+          const functionArgumentsCache = {};
+          context.functionArgumentsCache = functionArgumentsCache;
+
+          const bindingsStream = await engine.queryBindings(query, context);
+          expect((await bindingsStream.toArray()).map(res => res.get(DF.variable('len'))!.value)).toEqual(
+            quads.map(q => String(q.object.value.length)),
+          );
+          expect(Object.keys(functionArgumentsCache)).toContain('strlen');
+        });
+
+        it('is used when defaulted', async() => {
+          const alternativeEngine = new QueryEngine();
+          const context = <any> { sources: [ store ]};
+
+          // This cumbersome way is needed because evaluating with mock with isCalledWith gives you the reference
+          // to the cache and will make it seem like it was filled in from the beginning.
+          const original_function = (<any> alternativeEngine).actorInitQuery.mediatorContextPreprocess.mediate;
+          let firstFuncArgCache: object | undefined;
+          let secondFuncArgCache: object | undefined;
+          (<any> alternativeEngine).actorInitQuery.mediatorContextPreprocess.mediate = jest.fn(
```

Comment:
Browser tests falen door gebruik van `jest`.
Ik vermoed dat we dit gewoon kunnen vervangen door een gewone lambda?

Reply:
Thank you!